### PR TITLE
[TFS]: Removing old functions

### PR DESCRIPTION
**Pull request by Costallat**
Update Lua codebase to use new (meta-table) functions

Signed-off-by: Renato Foot Guimarães Costallat <costallat@hotmail.com>

### CODE REVIEW

**Comment by yatozin** [comment]
this breaks the being able to use bank money to pay for syuf at npcs

**Comment by Costallat** [comment]
Restored that option, please test again, tks!

**Comment by yatozin** [comment]
now it says " you do not have enough capacity" cant buy still`Lua Script Error: [Npc interface]
(Unknown scriptfile)
LuaScriptInterface::luaPlayerAddItemEx(). Item not found
stack traceback:
        [C]: in function 'addItemEx'
        data/npc/lib/npc.lua:57: in function 'doNpcSellItem'
        data/npc/lib/npcsystem/modules.lua:1062: in function 'callbackOnBuy'
        data/npc/lib/npcsystem/npchandler.lua:268: in function 'processModuleCallback'
        data/npc/lib/npcsystem/npchandler.lua:480: in function 'onBuy'
        data/npc/lib/npcsystem/modules.lua:1162: in function <data/npc/lib/npcsystem/modules.lua:1162>`

**Comment by Costallat** [comment]
Tks for testing!

You can try again now!

**Comment by yatozin** [comment]
'16:53 Sorc [500]: balance
16:53 Naji: You certainly have made a pretty penny. Your account balance is 982000 gold.
16:53 Sorc [500]: trade
16:53 Bought 1x bonelord shield for 7000 gold.
16:53 Bought 1x bonelord shield for 7000 gold.
16:53 Bought 1x bonelord shield for 7000 gold.
16:53 Bought 1x bonelord shield for 7000 gold.
16:53 Bought 1x bonelord shield for 7000 gold.
16:53 Bought 1x bonelord shield for 7000 gold.
16:53 Sorc [500]: balance
16:53 Naji: You certainly have made a pretty penny. Your account balance is 982000 gold.`

no money is taken from bank

**Comment by yatozin** [comment]
when you try buy exercise weapons in bags you get an empty brown bag now instead of shopping bag with the weapons.
also tibia 10 debugs on first login.

**Comment by Costallat** [comment]
Client 10 support will be dropped

**Comment by yatozin** [comment]
shopping bags come with no items. money is taken

**Comment by Costallat** [comment]
Still working on that, preparing the environment for test here too

**Comment by Costallat** [comment]
How did you test?

I tried with Nah'Bob and it worked, it just don't gave me any backpack, just the items, and the money was took from my account.

I think that other npc will have another flow.

Tks

**Comment by yatozin** [comment]
> How did you test?
> 
> I tried with Nah'Bob and it worked, it just don't gave me any backpack, just the items, and the money was took from my account.
> 
> I think that other npc will have another flow.
> 
> Tks

Xodet in thais
https://imgur.com/mE1fNWO

**Comment by Costallat** [comment]
Check now

**Comment by yatozin** [comment]
> Check now

cant see anymore problem at the moment.

File: data/creaturescripts/scripts/others/login.lua [hunk 1]
```diff
@@ -94,7 +94,52 @@ function onLogin(player)
             Unmute Player: / unmute nick.
             - Commands -]]
         player:popupFYI(msg)
-    end
+	end
+
```
Comment (miezaF) [inline data/creaturescripts/scripts/others/login.lua]: Make an option on config.lua
configKeys.ENABLE_QUEST_ACCESS = true, (false to disable, like global)
also add a Storage value to libs to give access only one time, if player doest have access yet

local strSto = player:getStorageValue(Storage.singleAccess)
if configManager.getNumber(configKeys.ENABLE_QUEST_ACCESS) and strSto < 1 then

File: data/creaturescripts/scripts/others/login.lua [hunk 2]
```diff
@@ -94,7 +94,52 @@ function onLogin(player)
...
+	player:setStorageValue(Storage.ForgottenKnowledge.AccessMachine, 1)
+	--SearoutsYalahar
+	player:setStorageValue(Storage.SearoutesAroundYalahar.Darashia, 1)
+	player:setStorageValue(Storage.SearoutesAroundYalahar.AbDendriel, 1)
+	player:setStorageValue(Storage.SearoutesAroundYalahar.Venore, 1)
+	player:setStorageValue(Storage.SearoutesAroundYalahar.Ankrahmun, 1)
+	player:setStorageValue(Storage.SearoutesAroundYalahar.PortHope, 1)
+	player:setStorageValue(Storage.SearoutesAroundYalahar.Thais, 1)
+	player:setStorageValue(Storage.SearoutesAroundYalahar.LibertyBay, 1)
+	player:setStorageValue(Storage.SearoutesAroundYalahar.Carlin, 1)
```
Comment (miezaF) [inline data/creaturescripts/scripts/others/login.lua]: end

Comment (miezaF) [inline data/creaturescripts/scripts/others/login.lua]: player:setStorageValue(Storage.singleAccess, 1)

**Review by miezaF — CHANGES_REQUESTED** [review]

**Comment by Costallat** [comment]
@yatozin can you do a final test now on client 10 and 12?

I restored the client 10 support that I removed, to keep it working until we reach some point where things get more complicated to keep the support of both versions.

Tks

**Comment by yatozin** [comment]
> @yatozin can you do a final test now on client 10 and 12?
> 
> I restored the client 10 support that I removed, to keep it working until we reach some point where things get more complicated to keep the support of both versions.
> 
> Tks

it all appears good

**Comment by majestyotbr** [comment]
Thanks for the contribution.

**Pull request merged**